demos/deblurring_demo/python/model.py

```diff
@@ -0,0 +1,107 @@
+        new_height = math.ceil(h / block_size) * block_size
+        new_width = math.ceil(w / block_size) * block_size
+
+        input_layer = next(iter(self.net.input_info))
+        input_shape = self.net.input_info[input_layer].input_data.shape
+        input_shape[2:] = (new_height, new_width)
+        self.net.reshape({input_layer: input_shape})
+
+    def prepare_inputs(self):
+        input_num = len(self.net.input_info)
+        if input_num != 1:
+            raise RuntimeError("Demo supports topologies only with 1 input")
+
+        input_blob_name = next(iter(self.net.input_info))
+        input_blob = self.net.input_info[input_blob_name]
+        input_blob.precision = "FP32"
+
+        input_size = input_blob.input_data.shape
+        if len(input_size) == 4 and input_size[1] == 3:
+            self.n, self.c, self.h, self.w = input_size
+        else:
+            raise RuntimeError("3-channel 4-dimensional model's input is expected")
+
+        return input_blob_name
+
+    def prepare_outputs(self):
+        output_num = len(self.net.outputs)
+        if output_num != 1:
+            raise RuntimeError("Demo supports topologies only with 1 output")
+
+        output_blob_name = next(iter(self.net.outputs))
+        output_blob = self.net.outputs[output_blob_name]
+        output_blob.precision = "FP32"
+
+        output_size = output_blob.shape
+        if len(output_size) == 4:
+            self.out_channels = output_size[1]
+            self.out_height = output_size[2]
+            self.out_width = output_size[3]
+        else:
+            raise Exception("Unexpected output blob shape {}. Only 4D output blob is supported".format(output_size))
+
+        return output_blob_name
+
+    def preprocess(self, inputs):
+        image = inputs
+        # right bottom padding to resize input image to input_layer shape
+        pad_params = {'mode': 'constant',
+                      'constant_values': 0,
+                      'pad_width': ((0, self.h - image.shape[0]), (0, self.w - image.shape[1]), (0, 0))
+                      }
+
+        if image.shape[0] != self.h or image.shape[1] != self.w:
+            resized_image = np.pad(image, **pad_params)
+        else:
+            resized_image = image
+        resized_image = resized_image.transpose((2, 0, 1))
+        resized_image = np.expand_dims(resized_image, 0)
+
+        meta = {'original_shape': image.shape,
+                'resized_shape': resized_image.shape}
+        dict_inputs = {self.input_blob_name: resized_image}
+        return dict_inputs, meta
+
+    def postprocess(self, outputs, meta):
+        prediction = outputs[self.output_blob_name].squeeze()
+        input_image_height = meta['original_shape'][0]
+        input_image_width = meta['original_shape'][1]
+
+        prediction = prediction.transpose((1, 2, 0))
+        prediction *= 255
+        prediction = prediction.astype(np.uint8)
+
+        return prediction[:input_image_height, :input_image_width, :]
```

Comment:
I think you should crop the image at first to minimize complexity of multiplying by 255 and type conversion.